File: ios/Plugin/InsomniaPlugin.m
#import <Foundation/Foundation.h>
#import <Capacitor/Capacitor.h>

// Define the plugin using the CAP_PLUGIN Macro, and
// each method the plugin supports using the CAP_PLUGIN_METHOD macro.
CAP_PLUGIN(InsomniaPlugin, "Insomnia",
           CAP_PLUGIN_METHOD(keepAwake, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(allowSleepAgain, CAPPluginReturnPromise);
)

File: ios/Plugin/InsomniaPlugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(InsomniaPlugin)
public class InsomniaPlugin: CAPPlugin {
    @objc func keepAwake(_ call: CAPPluginCall) {
        DispatchQueue.main.async {
            UIApplication.shared.isIdleTimerDisabled = true
            call.resolve()
        }
    }
    
    @objc func allowSleepAgain(_ call: CAPPluginCall) {
        DispatchQueue.main.async {
            UIApplication.shared.isIdleTimerDisabled = false
            call.resolve()
        }
    }
}
